src/main/java/com/example/crm_gym/repository/TraineeDAOImpl.java

```
@@ -0,0 +1,101 @@
+        try {
+            storage.update(traineeStorage, userId, trainee);
+            logger.info("Trainee updated successfully: {}", trainee);
+        } catch (Exception e) {
+            logger.error("Error updating trainee with userId: {}", userId, e);
+        } finally {
+            logger.info("Exiting update() with userId: {}", userId);
+        }
+    }
+
+    @Override
+    public void delete(int userId) {
+        logger.info("Entering delete() with userId: {}", userId);
+        try {
+            storage.remove(traineeStorage, userId);
+            logger.info("Trainee deleted successfully with userId: {}", userId);
+        } catch (Exception e) {
+            logger.error("Error deleting trainee with userId: {}", userId, e);
+        } finally {
+            logger.info("Exiting delete() with userId: {}", userId);
+        }
+    }
+
+    @Override
+    public Trainee findById(int userId) {
+        logger.info("Entering findById() with userId: {}", userId);
+        try {
+            Trainee trainee = storage.get(traineeStorage, userId);
+            if (trainee != null) {
+                logger.info("Trainee found: {}", trainee);
+            } else {
+                logger.warn("No trainee found with userId: {}", userId);
+            }
+            return trainee;
+        } catch (Exception e) {
+            logger.error("Error finding trainee with userId: {}", userId, e);
+            return null;
+        } finally {
+            logger.info("Exiting findById() with userId: {}", userId);
+        }
+    }
+
+    @Override
+    public List<Trainee> findAll() {
+        logger.info("Entering findAll()");
+        try {
+            List<Trainee> trainees = storage.findAll(traineeStorage);
+            logger.info("Returning all trainees, count: {}", trainees.size());
+            return trainees;
+        } catch (Exception e) {
+            logger.error("Error retrieving all trainees", e);
+            return null;
```

throw DaoException instead of returning value. And your service will catch DaoException and handle this case